addons/auth_totp_mail/models/auth_totp_device.py

```diff
@@ -28,13 +30,25 @@ def _generate(self, scope, name, expiration_date):
 
         res = super()._generate(scope, name, expiration_date)
 
-        self.env.user._notify_security_setting_update(
-            _("Security Update: Device Added"),
-            _(
-                "A trusted device has just been added to your account: %(device_name)s",
-                device_name=name
-            ),
+        message =  _(
+            "A trusted device has just been added to your account: %(device_name)s",
+            device_name=name,
         )
+        mail_values = None
+        if request:
+            # if the "New Connection" email has not been sent already,
+            # merge both to avoid spamming the user
+            new_connection_mail_mail_id = request.session.get('new_connection_mail_mail_id')
+            new_connection_mail = self.env['mail.mail'].browse(new_connection_mail_mail_id).sudo().exists()
+            if new_connection_mail and new_connection_mail.state == 'outgoing':
+                new_connection_mail.unlink()
+                del request.session['new_connection_mail_mail_id']
```

Comment:
Maybe we can pop it earlier before the if? Seems like if it doesn't exist or isn't outgoing we won't really ever care about it again right?